libp2p/transports/transport.nim

```
@@ -49,6 +55,8 @@ method start*(
   trace "starting transport on addrs", address = $addrs
   self.addrs = addrs
   self.running = true
+  if not isNil(self.onRunning):
+    self.onRunning.fire()
```

isn't `self.onRunning` always going to be initialized?

---

yes, thought the same thing. if for some reason it's not we don't want to silently ignore it.

---

Agreed. Removed the nil checks since they're unnecessary and would mask programming errors.